include/boost/math/differentiation/autodiff.hpp

```diff
@@ -271,6 +278,8 @@ class fvar
 
     explicit operator root_type() const; // Must be explicit, otherwise overloaded operators are ambiguous.
 
+    explicit operator int() const; // Must be explicit, otherwise overloaded operators are ambiguous.
```

Comment:
Wondering where this was required? (One place is enough.) If this was intentional by the boost::math developers, then the [Real Concepts](https://www.boost.org/doc/libs/1_69_0/libs/math/doc/html/math_toolkit/real_concepts.html) page needs to be updated to include it.

Reply:
The Airy and Bessel functions make calls to `boost::math::detail::iconv`, which uses `static_cast<int>` on type T if T isn't convertible to `int` (as determined by `boost::is_convertible`), otherwise `iround` would be used (but `iround` would need to be able to accept a policy argument).

This has been replaced with a generic conversion operator.

Reply:
> This has been replaced with a generic conversion operator.

I think that makes sense for integer types (`std::is_integral`), since the mathematical operation of truncation is locally constant, so any epsilon/derivative information would get wiped out. However for floating point types other than the root_type, I think it would be safer to give a compiler error. Otherwise it is effectively taking an autodiff type and silently replacing it with a constant (no derivative sensitivities.)

Reply:
I think that's reasonable, but investigating an issue when T is `float` (specific to `lambert_w`). There's an attempted cast from `double`; somehow, having root_type (`float`) and `int` conversion operators works, but root_type (`float`) and a generic integral conversion operator breaks (with a static assert).

Reply:
I reverted to using the previous conversion operators/struct (`root_type`, `int`, and an overloaded `real_cast`)